src/main/java/de/tum/in/www1/artemis/service/ProgrammingSubmissionService.java

```diff
@@ -697,11 +698,17 @@ protected Result lockSubmission(Submission submission, int correctionRound) {
         if (correctionRound == 0 && submission.getLatestResult() != null && AssessmentType.AUTOMATIC.equals(submission.getLatestResult().getAssessmentType())) {
             existingResult = submission.getLatestResult();
         }
+        else if (correctionRound == 0 && submission.getLatestResult() == null) {
+            // if the student did not participate in the exam, and therefore has no result
```

Comment:
It is a little bit confusing that you are talking about exams suddenly here. Maybe write something like: `This edge case can happen in the exam mode if the student did not participate ......`

Reply:
I changed the comment, as it turns out that wasn't the origin of the issue